tendrl/ceph_integration/sds_sync/__init__.py

```diff
@@ -67,17 +67,20 @@ def __init__(self):
     def _ping_cluster(self):
         NS.tendrl_context = NS.tendrl_context.load()
         NS.node_context = NS.node_context.load()
-        if NS.tendrl_context.cluster_id:
+        try:
+            if NS.tendrl_context.cluster_id:
                 cluster_data = ceph.heartbeat(NS.tendrl_context.cluster_id)
-                NS.tendrl_context.cluster_id = self.fsid = cluster_data['fsid']
-        else:
-            cluster_data = ceph.heartbeat()
-            if cluster_data:
+            else:
+                cluster_data = ceph.heartbeat()
+            if None not in cluster_data:
                 if "fsid" in cluster_data:
                     NS.tendrl_context.cluster_id = self.fsid = \
                         cluster_data['fsid']
-
-        NS.tendrl_context.cluster_name = self.name = cluster_data['name']
+                NS.tendrl_context.cluster_name = self.name = cluster_data['name']
+        except Exception as ex:
```

Comment:
What specific exception we expect here? Shouldn't we catch the specific one here?